ExtractArubaSW.ps1

```diff
@@ -0,0 +1,584 @@
+
+
+
+
+
+
+# Creation of the run variable
+
+$url = "rest/v4/ports"
+$response = invoke-ArubaSWWebRequest -method "GET" -url $url
+$run = ($response | convertfrom-json).port_element
+
+# Creation of the vlanports variable
+$vlanports = Get-ArubaSWVlansPorts
+$lldpremote = get-arubaswlldpremote
+
+$resultarray=@()
+
+# Crossing the information of the run variable against the vlanports variable
+
+foreach ($port in $run) {
+ 
+    $v = $vlanports | Where-Object {$_.port_id -eq $port.id}
+    $tagged=$null
+    $w = $lldpremote | Where-Object {$lldpremote.local_port -eq $port.id} 
+
+    $countvlan = -1
+        
+        foreach ($vlanp in $v) {
+            
+                $countvlan ++
+                
+                }
+
+        $comacountvlan = 0
+      
+        foreach ($vlanp in $v) {
+
+            
+            if ($comacountvlan -eq $countvlan) {
+
+                                                $coma = ''
+                                            
+                                                } else {
+                                            
+                                                $coma = ', '
+                                            
+                                                }
+            
+            if ($vlanp.port_mode -eq "POM_UNTAGGED") {
+            
+                    $untagged = $vlanp.vlan_id
+                    
+                    }
+
+            if ($vlanp.port_mode -eq "POM_TAGGED_STATIC") {
+
+                    $tagged += "$($vlanp.vlan_id)$coma"
+                    
+                    }
+
+            $comacountvlan++
+      
+        } 
+
+  
+    $item = New-Object PSObject
+    $item | Add-Member -type NoteProperty -Name 'name' -Value $($port.name)
+    $item | Add-Member -type NoteProperty -Name 'port_id' -Value $($port.id)
+    $item | Add-Member -type NoteProperty -Name 'port_tagged' -Value $($tagged)
+    $item | Add-Member -type NoteProperty -Name 'port_untagged' -Value $($untagged)
+    $item | Add-Member -type NoteProperty -Name 'lacp_status' -Value $($port.lacp_status)
+    $item | Add-Member -type NoteProperty -Name 'is_port_up' -Value $($port.is_port_up)
+    $item | Add-Member -type NoteProperty -Name 'lldp_port_id' -Value $($w.port_id)
+    $item | Add-Member -type NoteProperty -Name 'lldp_port_description' -Value $($w.port_description)
+     
+    $resultarray += $item
+
+}
+
+
+
+# Creation of Excel file
+
+
+$infoswitch = Get-ArubaSWSystemStatus
+
+
+
+
+# Launching an instance of MS Excel
+$excel = New-Object -ComObject "Excel.Application"            
+$excel.Visible = $True
+$excel.DisplayAlerts = $False
+
+# Creating an Excel sheet + activating the current sheet   
+$workbook = $excel.Workbooks.Add()
+$sheet = $workbook.Worksheets.Item(1)
+$sheet.Activate() | Out-Null
+
+
+
+# We position ourselves in A1 on Excel (line = 1 / column = 1)
+$row = 1
+$Column = 1
+$anum = 1
+$bnum = 1
+
+
+
+
+# Entering data in Excel
+
+ function ExcelFormatArray  {
```

Comment:
can use Powershell style cmdlet (like Add-ExcelFormatArray) you can use Get-Verbs to get list of valid verb for cmdlet

Reply:
done